server/server.go

```diff
@@ -435,7 +436,14 @@ func (a *ArgoCDServer) watchSettings() {
 			log.Infof("dex config modified. restarting")
 			break
 		}
-		if prevOIDCConfig != a.settings.OIDCConfigRAW {
+		newOIDCConfig := a.settings.OIDCConfig()
+		if prevOIDCConfig != nil && newOIDCConfig != nil {
+			if !reflect.DeepEqual(*prevOIDCConfig, *newOIDCConfig) {
```

Comment:
The previous `OIDCConfigRAW` comparison has better performance than using `reflect.DeepEqual`. I'm not sure I understand the reason why this code being changed here.

Reply:
Hi, thanks for the review. The previous OIDCConfigRAW as the name suggested is that it contains the OIDC config without the secret references being replaced with the actual secrets, e.g. `odic.config={client_secret='$k8s_secret_name:$k8s_secret_key'}. Therefore, if the actual underlying secrets change, argoCD does not restart to take these changes into account because the raw, non-interpolated config is still the same.